DataCapturing/Synchronization/Synchronizer.swift

```diff
@@ -122,12 +126,46 @@ public class Synchronizer {
     */
     public func syncChecked() {
         self.isReachableCheckingQueue.sync {
-            if isReachable(reachabilityManager!.networkReachabilityStatus) {
+            if syncOnWiFiOnly && isReachableOnEthernetOrWifi {
+                sync()
+            } else if isReachable {
                 sync()
             }
         }
     }
 
+    /**
+     Synchronize all measurements now.
+
+     The call is asynchronous, meaning it returns almost immediately, while the synchronization continues running inside its own thread.
+
+     You may call this method multiple times in short succession.
+     However only one synchronization can be active at a given time.
+     If you call this during an active synchronization it is going to return without doing anything.
+     */
+    public func sync() {
```

Comment:
This function is public now, so we can support our force sync button even under no WiFi conditions. Since it has become public the function has also moved up to the public functions.